template/src/utils/useUserList.ts

```diff
@@ -0,0 +1,24 @@
+/*
+********************************************
+ Copyright © 2021 Agora Lab, Inc., all rights reserved.
+ AppBuilder and all associated components, source code, APIs, services, and documentation 
+ (the “Materials”) are owned by Agora Lab, Inc. and its licensors. The Materials may not be 
+ accessed, used, modified, or distributed for any purpose without a license from Agora Lab, Inc.  
+ Use without a license or in violation of any license terms and conditions (including use for 
+ any purpose competitive to Agora Lab, Inc.’s business) is strictly prohibited. For more 
+ information visit https://appbuilder.agora.io. 
+*********************************************
+*/
+import {useContext} from 'react';
+import ChatContext from '../../src/components/ChatContext';
+
+/**
+ * This hook will fetch the user list
+ * @returns userList
+ */
+function useUserList() {
```

Comment:
Whereever you're trying to access or export a util to useContext, First Always validate it it's used within the provider,
```suggestion
 export function useUserList() {
   const context = useContext(ChatContext);
   if (context === undefined) {
     throw new Error(`useUserList must be used within a Chat Provider`);
   }
   return context;
 }

```

Reply:
useContext will return the default vaule. so we can't identify whether is wrapped with the provider or not. we will discuss on this one